### ISSUE TITLE
Does git-radar set background colors?

### ISSUE THREAD

**Issue by eproxus** [michaeldfallen/git-prompt#14]
I'm trying to replace my home grown version with `git-radar` but I'm having problems with background colors. Here's what my old theme looks like:

![](https://github.com/eproxus/pad.zsh-theme/raw/master/screenshot.png)

And here's what I see using `git-radar`:

<img width="561" alt="screen shot 2015-08-26 at 17 53 45" src="https://cloud.githubusercontent.com/assets/112878/9498715/70be20f8-4c1b-11e5-984c-cd08b633aff0.png">

Here's how I set the background and the text from `git-radar`:

```
echo "%{$BG[019]%}$(git-radar --zsh --fetch) "
```

So, it looks like the background is set on the branch name? Is this correct?

**Comment by eproxus** [comment]
Here's some "debug" output of what color codes `git-radar` generates in the above case, but I can't really figure out what they represent:

```
$ print -r -- ${(qqqq)$(git-radar --zsh --fetch)}
$'%{\033[01;30m%}' $'git:(%{\033[00m%}%{\033[37m%}master%{\033[00m%}%{\033[01;30m%})%{\033[00m%}' $'3%{\033[01;31m%}M%{\033[00m%}' $'1%{\033[01;37m%}A%{\033[00m%}'
```

**Comment by michaeldfallen** [comment]
It doesn't set background colours but it does presume that you don't. I use `$reset_color` to unset the foreground color but maybe it's also unsetting your background color? I'll have a look tonight and see if there's a reset foreground only option

**Comment by eproxus** [comment]
Yeah, that might be happening. Maybe there's a way to read the current color (both foreground and background) and then save it and reset to that? (I'm not very experienced with shell programming, so I have no idea if something like this exists).

**Comment by jalendport** [comment]
Any progress on this? I'm having the exact same problem.
